src/intro_movie_scenes_lib.c

```diff
@@ -0,0 +1,528 @@
+#include "global.h"
+#include "intro_movie_internal.h"
+#include "math_util.h"
+#include "system.h"
+#include "sys_task_api.h"
+#include "unk_0200FA24.h"
+
+void Task_IntroMovie_BlendFadeEffect(SysTask *task, void *pVoid);
+int IntroMovie_BgLayerToScrollEffectSlot(GFBgLayer bgId);
+void Task_IntroMovie_BgScroll_VBlank(SysTask *task, void *pVoid);
+void Task_IntroMovie_BgScroll_NotVBlank(SysTask *task, void *pVoid);
+void Task_IntroMovie_WindowPanEffect(SysTask *task, void *pVoid);
+void Task_IntroMovie_CircleWipeEffect(SysTask *task, void *pVoid);
+void IntroMovie_CircleWipeEffect_HBlankCB(void *pVoid);
+
+void IntroMovie_CreateSpriteResourceManagers(IntroMovieOverlayData *data, const u8 *counts) {
+    for (u8 i = 0; i < 4; ++i) {
+        data->spriteResManagers[i] = Create2DGfxResObjMan(counts[i], (GfGfxResType)i, HEAP_ID_INTRO_MOVIE);
+    }
+}
+
+void IntroMovie_DestroySpriteResourceManagers(IntroMovieOverlayData *data) {
+    for (u8 i = 0; i < 4; ++i) {
+        Destroy2DGfxResObjMan(data->spriteResManagers[i]);
+    }
+}
+
+GF_2DGfxResMan **IntroMovie_GetSpriteResourceManagersArray(IntroMovieOverlayData *data) {
+    return data->spriteResManagers;
+}
+
+void IntroMovie_StartSpriteAnimAndMakeVisible(Sprite *sprite, BOOL active) {
+    Set2dSpriteAnimActiveFlag(sprite, active);
+    Set2dSpriteVisibleFlag(sprite, active);
+}
+
+void IntroMovie_BuildSpriteResourcesHeaderAndTemplate(int resId, IntroMovieOverlayData *data, int priority, NNS_G2D_VRAM_TYPE whichScreen, SpriteTemplate *template, SpriteResourcesHeader *header) {
+    CreateSpriteResourcesHeader(
```

Comment:
as far as I'm concerned then, giving it some thought, everything else is fine and is mainly me being pedantic, however, I will say that I really *would* like this one function fixed, before approval, if you do that then I can approve, without it'll probably have to wait until the clang-format PR (which is in progress) to be merged imo

Reply:
I'm sorry but no, there's too many arguments to print them all on line line and still be readable.

Reply:
I mean that's ok but please split them over two lines then, with many on one line before it becomes too much, then more on the second etc. please do not have one line per param

Reply:
again, I'm going to fight you on this. one line per param is preferable in this case.